graphql/util.go

```diff
@@ -909,6 +919,53 @@ func getHostRequestOptions(ctx context.Context, usr *user.DBUser, spawnHostInput
 	return options, nil
 }
 
+// GetProjectAndModulesForTask returns the project ref and modules for a task
+func getProjectAndModulesForTask(ctx context.Context, settings *evergreen.Settings, t *task.Task) (*model.ProjectRef, []manifest.Module, error) {
+	if t == nil {
+		return nil, nil, errors.New("task is nil")
+	}
+	projectRef, err := model.FindBranchProjectRef(t.Project)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "finding project ref")
+	}
+
+	v, err := model.VersionFindOne(model.VersionById(t.Version))
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "finding version")
+	}
+	pp, err := model.ParserProjectFindOneByID(ctx, settings, v.ProjectStorageMethod, v.Id)
+	if err != nil || pp == nil {
+		return nil, nil, errors.Wrap(err, "finding project ref")
+	}
+	// Then fetch each of the modules
+	project, err := model.TranslateProject(pp)
+	if err != nil || project == nil {
+		return nil, nil, errors.Wrap(err, "translating project")
+	}
+	variant := project.FindBuildVariant(t.BuildVariant)
+	if variant == nil {
+		return nil, nil, errors.Wrapf(err, "finding build variant '%s' in config", t.BuildVariant)
+	}
+
+	var manifestModules []manifest.Module
```

Comment:
Why are we using the manifest modules instead of the project modules? Then we wouldn't need any of this name stuff.